### ISSUE TITLE
Error ORA-03114 occurs during all tables export starting from 17.5 version

### ISSUE THREAD

**Issue by rautars** [darold/ora2pg#634]
Hello,
As a developer I wanted prepare application backend support for PostgreSQL database in addition of existing Oracle. Development is done on Windows PC and as there is easier to install ora2pg on linux env I decided to use Docker.

Here Dockerfile
```
FROM fedora

MAINTAINER xeranas <xeranas@somedomain.com> 

RUN dnf update -y && dnf install unzip gcc perl perl-DBI perl-CPAN tar redhat-rpm-config perl-DBD-Pg postgresql-devel perl-open -y

RUN mkdir /usr/lib/oracle/11.2/client64/network/admin -p

# Download RPM files from http://www.oracle.com/technetwork/topics/linuxx86-64soft-092277.html
ADD oracle-instantclient11.2-basic-11.2.0.4.0-1.x86_64.rpm /tmp/ora-basic.rpm
ADD oracle-instantclient11.2-devel-11.2.0.4.0-1.x86_64.rpm /tmp/ora-devel.rpm
ADD oracle-instantclient11.2-sqlplus-11.2.0.4.0-1.x86_64.rpm /tmp/ora-sqlplus.rpm

RUN dnf install /tmp/ora-basic.rpm --nogpgcheck -y \
	&& rm -f /tmp/ora-basic.rpm \
	&& dnf install /tmp/ora-devel.rpm --nogpgcheck -y \
	&& rm -f /tmp/ora-devel.rpm \ 
	&& dnf install /tmp/ora-sqlplus.rpm --nogpgcheck -y \ 
	&& rm -f /tmp/ora-sqlplus.rpm

ENV ORACLE_HOME=/usr/lib/oracle/11.2/client64
ENV TNS_ADMIN=/usr/lib/oracle/11.2/client64/network/admin
ENV LD_LIBRARY_PATH=/usr/lib/oracle/11.2/client64/lib
ENV PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/lib/oracle/11.2/client64/bin

RUN curl -L http://www.perl.org/CPAN/authors/id/P/PY/PYTHIAN/DBD-Oracle-1.74.tar.gz | (cd /tmp && tar -zxvf -) && mv /tmp/DBD-Ora* /tmp/DBD-Oracle
RUN cd /tmp/DBD-Oracle && perl Makefile.PL -l && make && make install

RUN mkdir /data

VOLUME /data

RUN curl -L -o /tmp/ora2pg.zip https://github.com/darold/ora2pg/archive/v17.5.zip && (cd /tmp && unzip ora2pg.zip && rm -f ora2pg.zip) && mv /tmp/ora2pg* /tmp/ora2pg && (cd /tmp/ora2pg && perl Makefile.PL && make && make install)

```

I get error
```
FATAL: ORA-03114: not connected to ORACLE (DBD ERROR: OCIStmtExecute/Describe)
Aborting export...
```

version check via `ora2pg -c config/ora2pg.conf -t SHOW_VERSION` returns `Oracle Database 11g Enterprise Edition Release 11.2.0.4.0` within few seconds

If I change ora2pg version on Dockerfile (in last command) from v17.5.zip to v17.4.zip and build new image I get no errors when using same ora2pg.conf. Same issue occurs also on 18.2 version.

I can successfully export views (there only few) and single table (by defining it in ALLOW). However it does seems cannot export all tables at once at least in same manner as it was able in 17.4 version which using same database instance was able to export around 1300 tables.

**Comment by darold** [comment]
Does this issue still occurs? In this case do you have an up to date Dockerfile to try to reproduceo should I use this one?

**Comment by xeranas** [comment]
I only tried released version 18.2 and down to 17.4 (each release version). 17.4 was last version which worked for my environment best. As I was manage to export object (views, tables) when their number is smaller. Oracle server which I do connect is on remote computer. 

Last lines of docker file defines version of ora2pg, only curl url need to be replaced for different ora2pg version:

```
RUN curl -L -o /tmp/ora2pg.zip https://github.com/darold/ora2pg/archive/v17.5.zip ...
```

**Comment by janteque** [comment]
@darold I'm facing exactly the same problem with version 23.1

 However, using version 17.4 works fine. Any idea of the reasons behind?
